core/src/stack/outputs.rs

```diff
@@ -218,22 +222,34 @@ impl ToElements<Felt> for StackOutputs {
 
 impl Serializable for StackOutputs {
     fn write_into<W: ByteWriter>(&self, target: &mut W) {
-        debug_assert!(self.stack.len() <= u32::MAX as usize);
-        target.write_u32(self.stack.len() as u32);
+        debug_assert!(self.stack.len() <= MAX_STACK_OUTPUTS_SIZE);
+        target.write_usize(self.stack.len());
         target.write_many(&self.stack);
 
-        debug_assert!(self.overflow_addrs.len() <= u32::MAX as usize);
-        target.write_u32(self.overflow_addrs.len() as u32);
+        debug_assert!(self.overflow_addrs.len() <= MAX_OVERFLOW_ADDRS_NUM);
+        target.write_usize(self.overflow_addrs.len());
         target.write_many(&self.overflow_addrs);
```

Comment:
Thinking about this more, output stack depth and the number of overflow addresses should be related to each other. That is, by knowing stack depth, we can figure out the number of overflow addresses (i think it is something like `stack_depth - 16`).

So, can probably serialize just the stack depth, and then use it to read both vectors.